Mathlib/LinearAlgebra/Matrix/Nondegenerate.lean

```diff
@@ -20,16 +20,23 @@ import Mathlib.LinearAlgebra.Matrix.Adjugate
 
 namespace Matrix
 
-variable {m R A : Type*} [Fintype m] [CommRing R]
+variable {m R A : Type*} [CommRing R]
 
 /-- A matrix `M` is nondegenerate if for all `v ≠ 0`, there is a `w ≠ 0` with `w * M * v ≠ 0`. -/
-def Nondegenerate (M : Matrix m m R) :=
+def Nondegenerate [Finite m] (M : Matrix m m R) :=
+  let _i : Fintype m := Fintype.ofFinite m
```

Comment:
```suggestion
  letI : Fintype m := Fintype.ofFinite m
```
inlining it should give a cleaner term. For defs, this can be relevant.

Reply:
Done. Though I claim that this definition should never be unfolded since we provide `nondegenerate_def` (which is of course the one place it is unfolded).